File: LeetCode/102/main.py
"""
DIFFICULTY : medium
TAGS : tree, bfs, binary tree
"""
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        depth = 0
        queue = []
        queue.append(root)
        ans = []
        while queue:
            # depth += 1
            temp = []
            for _ in range(len(queue)):
                node = queue.pop(0)
                if node:
                    temp.append(node.val)
                    if node.left:
                        queue.append(node.left)
                    if node.right:
                        queue.append(node.right)
            ans.append(temp)

        return ans

File: leetcode/102/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {

	if root == nil {
		return [][]int{}
	}

	ans := [][]int{}

	q := []*TreeNode{root}

	for len(q) != 0 {
		size := len(q)
		level := []int{}
		for i := 0; i < size; i++ {
			node := q[0]
			level = append(level, node.Val)
			q = q[1:]

			if node.Left != nil {
				q = append(q, node.Left)
			}
			if node.Right != nil {
				q = append(q, node.Right)
			}
		}
		ans = append(ans, level)
	}
	return ans
}
